src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +263,216 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+            {
+                variantDefinition = ResolveDefaultVariantDefinition(featureDefinition, isFeatureEnabled);
+            }
+
+            return variantDefinition;
+        }
+
+        private ValueTask<VariantDefinition> AssignVariantAsync(FeatureDefinition featureDefinition, TargetingContext targetingContext, CancellationToken cancellationToken)
+        {
+            VariantDefinition variant = null;
+
+            if (featureDefinition.Allocation.User != null)
+            {
+                foreach (UserAllocation user in featureDefinition.Allocation.User)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext, user.Users, _assignerOptions.IgnoreCase))
+                    {
+                        variant = featureDefinition.Variants.FirstOrDefault((variant) => variant.Name == user.Variant);
+
+                        if (!string.IsNullOrEmpty(variant.Name))
+                        {
+                            return new ValueTask<VariantDefinition>(variant);
+                        }
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Group != null)
+            {
+                foreach (GroupAllocation group in featureDefinition.Allocation.Group)
+                {
+                    if (TargetingEvaluator.IsGroupTargeted(targetingContext, group.Groups, _assignerOptions.IgnoreCase))
+                    {
+                        variant = featureDefinition.Variants.FirstOrDefault((variant) => variant.Name == group.Variant);
+
+                        if (!string.IsNullOrEmpty(variant.Name))
+                        {
+                            return new ValueTask<VariantDefinition>(variant);
+                        }
+                    }
+                }
+            }
+
+            if (featureDefinition.Allocation.Percentile != null)
+            {
+                foreach (PercentileAllocation percentile in featureDefinition.Allocation.Percentile)
+                {
+                    if (TargetingEvaluator.IsTargeted(targetingContext, percentile.From, percentile.To, _assignerOptions.IgnoreCase, featureDefinition.Allocation.Seed))
+                    {
+                        variant = featureDefinition.Variants.FirstOrDefault((variant) => variant.Name == percentile.Variant);
+
+                        if (!string.IsNullOrEmpty(variant.Name))
+                        {
+                            return new ValueTask<VariantDefinition>(variant);
+                        }
+                    }
+                }
+            }
+
+            return new ValueTask<VariantDefinition>(variant);
+        }
+
+        private VariantDefinition ResolveDefaultVariantDefinition(FeatureDefinition featureDefinition, bool isFeatureEnabled)
+        {
+            string defaultVariantPath = isFeatureEnabled ? featureDefinition.Allocation.DefaultWhenEnabled : featureDefinition.Allocation.DefaultWhenDisabled;
```

Comment:
why is that called path and not name?

Reply:
Missed this when changing the code I think, name is right